Cleanup

siblun opened the pull request:
Clean up dependencies.

siblun committed:
Update requirements.txt

shkolyaric commented:
1. retry logic is overhead [there](https://github.com/siblun/clickhouse-data-generator/blob/fd3fdc3fce1ad786aef0cf880a055026639792cc/src/main.py#L55)
2. we must remove reading table scheme from. This fixture only make project ugly.  
3. some output still only for some kind of people [there](https://github.com/siblun/clickhouse-data-generator/blob/fd3fdc3fce1ad786aef0cf880a055026639792cc/src/main.py#L139). pls translate text and use `logging` instead of `print`

siblun committed:
Feat: Introduce generate_rows_batch method in DataGenerator

Created a new method `generate_rows_batch` within the `DataGenerator` class to encapsulate the logic for generating batches of rows.

siblun committed:
Refactor: Cleaned up and improved main data script

This update makes `main.py` better by:
- Using English logging for all messages.
- Taking out the extra retry steps for getting table schema.
- Using the new batch generation method from `data_generator.py`.
- Making sure the table schema is always read from ClickHouse, with the SQL file only for initial table setup.

siblun committed:
Docs: Translate clickhouse_client.py documentation to English

siblun committed:
Docs: Translate config_parser.py documentation to English

siblun committed:
Docs: Translate data_generator.py documentation to English

siblun committed:
Refactor: SchemaParser now fetches schema only from ClickHouse

Removed the `parse_schema_from_sql_file` method.

siblun committed:
Feat: Add table_definition retrieval to ConfigParser

Added a new method `get_table_definition` to ConfigParser to allow retrieving
the full table schema directly from the config.json file.

siblun committed:
Fix: Add inline table definition

siblun committed:
Refactor: Streamlined main script with dynamic table creation and fixes

siblun committed:
Update README.md

siblun committed:
Delete schemas directory

siblun committed:
feat: Require pre-existing tables for data insertion

siblun committed:
Update README.md

shkolyaric commented on src/main.py:
```diff
@@ -1,9 +1,21 @@
+This script performs the following steps:
+1. Reads configuration from the `config.json` file.
+2. Establishes a connection to ClickHouse.
+3. Dynamically creates the target table from definition in config.json if specified,
```
we must not create table, only work with already existing one

shkolyaric commented on src/main.py:
```diff
@@ -48,95 +60,42 @@ def main():
     except FileNotFoundError as e:
         logging.error("Critical error: File not found. %s", e)
-        print("Пожалуйста, убедитесь, что файл конфигурации и/или файл схемы существуют по указанным путям.")
+        logging.error("Please ensure the configuration file and/or schema file exist at the specified paths.")
```
we do not process schema file

shkolyaric commented on src/main.py:
```diff
@@ -32,7 +44,7 @@ def main():
 
         table_name = table_info['name']
-        schema_file_path = table_info['schema_file_path']
+        schema_file_path = table_info.get('schema_file_path')
```
we removed schema file support

siblun committed:
Refactor: Cleanup main.py

siblun committed:
Refactor: Remove table creation related settings from ConfigParser

shkolyaric commented on src/main.py:
```diff
@@ -48,95 +59,42 @@ def main():
-            if len(current_batch) >= inserts_per_query or (i == total_inserts - 1 and current_batch):
+            current_batch = data_generator.generate_rows_batch(rows_to_generate_in_batch)
+
+            if current_batch:
```
check `current_batch` is really necessary?

siblun committed:
Refactor: Remove redundant 'if current_batch' check

Removed the unnecessary 'if current_batch:' condition in the data insertion loop
within `main.py`. The `data_generator.generate_rows_batch()` method is guaranteed
to return a non-empty list when `rows_to_generate_in_batch` is greater than zero,
making the check redundant.

siblun committed:
Refactor: Remove unused imports in schema_parser.py
